src/Visitor.ts

```diff
@@ -0,0 +1,234 @@
+              range,
+              symbol: this.lsifSymbol(symbol).value,
+            })
+          )
+        }
+      }
+    }
+  }
+
+  private addSymbolInformation(
+    node: ts.Node,
+    sym: ts.Symbol,
+    symbol: LsifSymbol
+  ): void {
+    this.document.symbols.push(
+      new lsif_typed.SymbolInformation({
+        symbol: symbol.value,
+        documentation: [
+          '```ts\n' +
+            this.checker.typeToString(this.checker.getTypeAtLocation(node)) +
+            '\n```',
+          ts.displayPartsToString(sym.getDocumentationComment(this.checker)),
+        ],
+      })
+    )
+  }
+
+  private declarationName(node: ts.Node): ts.Node | undefined {
+    if (
+      ts.isEnumDeclaration(node) ||
+      ts.isVariableDeclaration(node) ||
+      ts.isPropertyDeclaration(node) ||
+      ts.isMethodSignature(node) ||
+      ts.isMethodDeclaration(node) ||
+      ts.isPropertySignature(node) ||
+      ts.isFunctionDeclaration(node) ||
+      ts.isModuleDeclaration(node) ||
+      ts.isPropertyAssignment(node) ||
+      ts.isShorthandPropertyAssignment(node) ||
+      ts.isParameter(node) ||
+      ts.isTypeParameterDeclaration(node) ||
+      ts.isInterfaceDeclaration(node)
+    ) {
+      return node.name
+    }
+    return undefined
+  }
+
+  private lsifSymbol(node: ts.Node): LsifSymbol {
+    const fromCache: LsifSymbol | undefined =
+      this.globalSymbolTable.get(node) || this.localSymbolTable.get(node)
+    if (fromCache) {
+      return fromCache
+    }
+    if (ts.isBlock(node)) {
+      return LsifSymbol.empty()
+    }
+    if (ts.isSourceFile(node)) {
+      const package_ = this.packages.symbol(node.fileName)
+      if (!package_) {
+        return this.cached(node, LsifSymbol.empty())
+      }
+      return this.cached(node, package_)
+    }
+    if (
+      ts.isPropertyAssignment(node) ||
+      ts.isShorthandPropertyAssignment(node)
+    ) {
+      const name = node.name.getText()
+      let counter = this.propertyCounters.get(name)
+      if (!counter) {
+        counter = new Counter()
+        this.propertyCounters.set(name, counter)
+      }
+      return this.cached(
+        node,
+        LsifSymbol.global(
+          this.lsifSymbol(node.getSourceFile()),
+          Descriptor.meta(`${node.name.getText()}${counter.next()}`)
+        )
+      )
+    }
+    const owner = this.lsifSymbol(node.parent)
+    if (owner.isEmptyOrLocal()) {
+      return this.newLocalSymbol(node)
+    }
+
+    if (isAnonymousContainerOfSymbols(node)) {
+      return this.cached(node, this.lsifSymbol(node.parent))
+    }
+    if (ts.isImportSpecifier(node)) {
+      const tpe = this.checker.getTypeAtLocation(node)
+      for (const declaration of tpe.symbol?.declarations || []) {
+        return this.lsifSymbol(declaration)
+      }
+    }
+
+    const desc = this.descriptor(node)
+    if (desc) {
+      return this.cached(node, LsifSymbol.global(owner, desc))
+    }
+    return this.newLocalSymbol(node)
```

Comment:
I don't fully understand this last return. I see it being hit in a few cases when the tests are run:

```
hit last code path for a with parent { a }
  hit last code path for (param: string) => string with parent methodSignature2: (param: string) => string
hit last code path for (param: string): string => {
      return param
    } with parent methodSignature2: (param: string): string => {
      return param
    }
```

Why are these cases not being caught earlier with the `owner.isEmptyOrLocal()` check?

Reply:
This case happens for local symbols that have a global owner. For example, blocks `{}` have a symbol encoding in the compiler and they can be direct children of exported functions.